### PR TITLE
Deploy RMO on Hypershift management clusters

### PR DESCRIPTION
**Pull request by sam-nguyen7**
Update OLM template to deploy RMO on management clusters.
https://issues.redhat.com/browse/OSD-14461

### CODE REVIEW

**Review by rafael-azevedo — APPROVED** [review]
/lgtm

File: hack/olm-registry/olm-artifacts-template.yaml [hunk 1]
```diff
@@ -1,5 +1,7 @@
 apiVersion: v1
 kind: Template
+metadata: selectorsyncset-template
```
Comment (mjlshen) [inline hack/olm-registry/olm-artifacts-template.yaml:3]: I think this should still be `.metadata.name` right?

```suggestion
metadata:
  name: selectorsyncset-template
```

**Review by mjlshen — APPROVED** [review]
/unhold
/lgtm

**Review by rafael-azevedo — APPROVED** [review]
/lgtm

**Pull request merged**